src/CommunityVoices/App/Api/Component/FileProcessor.php

```diff
@@ -0,0 +1,207 @@
+           $sheetData = [];
+           if (empty($columnNameErrors))  {
+              fgetcsv($f);
+               while (($data = fgetcsv($f)) !== FALSE) {
+                   $dataToAdd = ['rowData' => []];
+                   $identifier = false;
+                   for ($i = 0; $i < count($columnOrder); $i++) {
+                       $columnName = $columnOrder[$i]; // XML requires no spaces
+                       $currentColumnData = $this->replaceTextInTags($data[$i]);
+                       if($columnName != "unrecognized") {
+                           $originalName = $sourceNameMapper[$columnName];
+                           if($columnName=="identifier") {
+                               $identifier = "i" . $this->cleanString($currentColumnData); // identifier must start with a letter for xml parse so this takes care of that
+                               if(!empty($identifier)) array_push($validIdentifiers["allIdentifiers"],["item" => $identifier]);
+                           } else {
+                               $dataToAdd['rowData'][$columnName] = ["originalName" => $originalName, "columnData" => $currentColumnData, "formattedName" => $columnName];
+                           }
+                       }
+                   }
+                   if($identifier) {
+                       $dataToAdd['rowData']["attribution"]["error"] = self::ERR_MISSING_ATTRIBUTION;
+                       $sheetData[$identifier] = $dataToAdd;
+                       $sheetData[$identifier]["quotes"] = []; // allows quotes related to source info
+                   }
+               }
+           }
+
+           fclose($f);
+        }
+
+         if (($f= fopen($quoteFilePath, "r")) !== FALSE) {
+             $columnOrder = []; // used to track column locations since we are going entirely by name instead of order
+             $givenColumnNames = fgetcsv($f);
+             foreach ($givenColumnNames as $column) {
+                 $formattedColumn = $this->cleanString($column);
+                 if(in_array($formattedColumn,$formattedQuoteNames)) {
+                     array_push($columnOrder,$formattedColumn);
+                 } else {
+                     array_push($columnOrder,"unrecognized");
+                     array_push($columnNameWarnings["unrecognized"],["item" => [$column]]);
+                 }
+             }
+
+
+             if(!in_array("contentcategory1",$columnOrder)) array_push($columnNameErrors,["item" => [self::ERR_NO_CONTENT_CATEGORIES]]);
+             if(!in_array("editedquotes",$columnOrder)) array_push($columnNameErrors,["item" => [self::ERR_NO_QUOTE]]);
+             if(!in_array("identifier",$columnOrder)) array_push($columnNameErrors,["item" => [self::ERR_NO_IDENTIFIER]]);
+             // These are both major errors. There is no need to parse the rest of the sheet as uploading will not be allowed if one of these errors occurs
```

Comment:
Again, a bit confused about this comment placement.

Reply:
@Sammidysam The word "both" is misleading, It should say "all three of these are major errors". I am trying to point out that if the user is lacking essential information this will require a reupload, so no need to parse rest of sheet. Does that make sense or are you still confused?

Reply:
If you can change the wording so that it lines up with there being 3 cases, that would make it make total sense.

Reply:
Fixed